### PR TITLE
Update core API docs: Adapter and Model

### PR DESCRIPTION
**Pull request by sovrasov**
# What does this PR do?
Update the inline docs for adapter and model.

### CODE REVIEW

File: model_api/python/model_api/adapters/inference_adapter.py [hunk 1]
```diff
@@ -122,11 +124,11 @@ def infer_async(self, dict_data, callback_data):
         """
 
     @abstractmethod
-    def get_raw_result(self, infer_result) -> dict:
+    def get_raw_result(self, infer_result: dict) -> dict:
         """Gets raw results from the internal inference framework representation as a dict.
 
         Args:
-            - infer_result: framework-specific result of inference from the model
+            - infer_resul (dict): framework-specific result of inference from the model
```
Comment (ashwinvaidya17) [inline model_api/python/model_api/adapters/inference_adapter.py:131]:
```suggestion
            - infer_result (dict): framework-specific result of inference from the model
```

File: model_api/python/model_api/models/model.py [hunk 2]
```diff
@@ -402,11 +422,24 @@ def batch_infer_callback(result, id):
...
             self.model_loaded = True
             self.inference_adapter.load_model()
 
-    def reshape(self, new_shape):
+    def reshape(self, new_shape: dict):
+        """
+        Reshapes the model inputs to fit the new input shape.
+
+        Args:
+            new_shape (_type_): a dictionary with inputs names as keys and
```
Comment (ashwinvaidya17) [inline model_api/python/model_api/models/model.py:440]:
```suggestion
            new_shape (dict): a dictionary with inputs names as keys and
```

**Review by ashwinvaidya17 — COMMENTED** [review]
Thanks looks good. Two very minor comments

**Comment by sovrasov** [comment]
> Thanks looks good. Two very minor comments

Thanks, applied your suggestions

**Review by ashwinvaidya17 — APPROVED** [review]

**Pull request merged**